allennlp/data/vocabulary.py

```diff
@@ -604,21 +610,24 @@ def save_to_files(self, directory: str) -> None:
         if os.listdir(directory):
             logging.warning("vocabulary serialization directory %s is not empty", directory)
 
-        with codecs.open(
-            os.path.join(directory, NAMESPACE_PADDING_FILE), "w", "utf-8"
-        ) as namespace_file:
-            for namespace_str in self._non_padded_namespaces:
-                print(namespace_str, file=namespace_file)
-
-        for namespace, mapping in self._index_to_token.items():
-            # Each namespace gets written to its own file, in index order.
+        # We use a lock file to avoid race conditions where multiple processes
+        # might be reading/writing from/to the same vocab files at once.
+        with FileLock(os.path.join(directory, ".lock")):
```

Comment:
Does this mean you can remove the barrier in the training loop?

Reply:
Yea I think we can